tests/clock/ClockUtils.py

```diff
@@ -0,0 +1,384 @@
+                pytest_assert(False, f'Given string "{date_str}" is not a valid date')
+
+    @staticmethod
+    def validate_time(time_str):
+        """
+        @summary:
+            Verify that given string is in a good time format: "11:29:46 AM" (or PM)
+        @param time_str: the given string
+        """
+        with allure.step(f'Validate time for: "{time_str}"'):
+            logging.info(f'Validate time for: "{time_str}"')
+            try:
+                datetime.datetime.strptime(time_str, "%I:%M:%S %p")
+                logging.info('Validate time success')
+            except ValueError:
+                logging.info('Validate time fail')
+                pytest_assert(False, f'Given string "{time_str}" is not a valid time')
+
+    @staticmethod
+    def get_valid_timezones(duthosts):
+        """
+        @summary:
+            Get the list of valid timezones from 'show clock timezones' command
+        @param duthosts: duthosts object
+        @return: list of timezones (strings)
+        """
+        with allure.step('Get list of valid timezones from show clock timezones command'):
+            logging.info('Get list of valid timezones from show clock timezones command')
+            return ClockUtils.run_cmd(duthosts=duthosts, cmd=ClockConsts.CMD_SHOW_CLOCK_TIMEZONES).split()
+
+    @staticmethod
+    def validate_timezone(timezone_str, duthosts):
+        """
+        @summary:
+            Verify that the given string is an abbreviation of a valid timezone
+        @param timezone_str: the given string
+        @param duthosts: duthosts
+        """
+        with allure.step(f'Verify that given string "{timezone_str}" is a valid timezone'):
+            logging.info(f'Verify that given string "{timezone_str}" is a valid timezone')
+
+            with allure.step('Get timezone from timedatectl linux command'):
+                logging.info('Get timezone from timedatectl linux command')
+                timedatectl_output = \
+                    ClockUtils.parse_linux_cmd_output(ClockUtils.run_cmd(duthosts, ClockConsts.CMD_TIMEDATECTL))
+                timedatectl_timezone = timedatectl_output[ClockConsts.TIME_ZONE]
+                timedatectl_tz_name = timedatectl_timezone.split()[0]
+                timedatectl_tz_abbreviation = timedatectl_timezone.split(' ')[1].split(',')[0].replace('(', '')
+                logging.info(f'Timezone in timedatectl: "{timedatectl_timezone}"\nTimezone name: '
+                             f'"{timedatectl_tz_name}"\nTimezone abbreviation: "{timedatectl_tz_abbreviation}" ')
+
+            with allure.step(f'Verify timezone name "{timedatectl_tz_name}" from timedatectl is in valid timezones'):
+                logging.info(f'Verify timezone name "{timedatectl_tz_name}" from timedatectl is in valid timezones')
+                valid_timezones = ClockUtils.get_valid_timezones(duthosts)
+                pytest_assert(timedatectl_tz_name in valid_timezones,
```

Comment:
There is one specific extra timezone added by the clock commands HLD apart from the ones in timedatectl command. (Etc/UTC). This should be taken account into consideration here.

Reply:
when 'Etc/UTC' is configured, 'show clock' output ends with the timezone abbreviation 'UTC'.
in this case, when we run 'show clock' and validate the timezone in it ('UTC'), we run 'timedatectl', which shows 'Time zone: Etc/UTC (UTC, +0000)' . we extract the 'Etc/UTC' from the 'timedatectl' output, verify it's in valid timezones (received it from 'show clock timezones'), and it is in our case.
after that, we take the 'UTC' out of the timezone value from 'timedatectl' output, and just verify that it's equal to the param. 
![image](https://github.com/sonic-net/sonic-mgmt/assets/33894564/c1a8f574-d172-4cfa-bba6-00d72a464e79)

Reply:
after some changes, this method is no longer needed. resolving this comment